File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Horario.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalTime;

@Entity
@Table(name ="HORARIO")
@Getter
@Setter
@NoArgsConstructor
public class Horario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Cod_horario")
    private Integer codHorario;

    @ManyToOne
    @JoinColumn(name = "Cod_ClaseSkate_FK")
    private ClaseSkate claseSkate;

    @Column(name="Dia")
    private String dia;

    @Column(name="Hora")
    private LocalTime hora;
}

File: sbschoolmanager_app/lib/models/horario.dart

import "clase_skate.dart";

class Horario {

  final int? codHorario;
  final ClaseSkate claseSkate;
  final String dia;
  final String hora;

  Horario({
    this.codHorario,
    required this.claseSkate,
    required this.dia,
    required this.hora,

  });

  factory Horario.fromJson(Map<String, dynamic> json){
    return Horario(
      codHorario: json["codHorario"],
      claseSkate: ClaseSkate.fromJson(json["claseSkate"]),
      dia: json["dia"],
      hora: json["hora"],
    );
  }

  Map<String, dynamic> toJson(){
    return{
      if(codHorario != null) 
      "codHorario": codHorario,
      "claseSkate": claseSkate.toJson(),
      "dia": dia,
      "hora": hora,
    };
  }

}